### ISSUE TITLE
--delete not removing directories

### ISSUE THREAD

**Issue by vxbinaca** [jjjake/internetarchive#225]
I did a [mirror of a apache directory](https://archive.org/details/lftp-mirror-mardenz-SVG) with LFTP and managed to upload the contents after a few tries, however it's left the directory structure in-tact on my machine even though I used `--delete`.

**Comment by androclus** [comment]
This also presents itself in another more pernicious form:

Apparently, when ia delete is run, archive.org creates a new "history" subfolder, and puts stuff in there. (I would really prefer it not do this, but there appears to be no way to prevent it.)

Problems arise:

1. When then uploading new files (to replace the old ones) and deriving them, archive.org still sees and picks up the files in the history sub-folder (which, theoretically, could be considered a bug by itself, yes?)

2. Furthermore, there is a problem right now with the edit-files page, where Flash has been removed, but there is not yet a good interface to replace it. (JK has told me this is being worked on.) So it has a bug, in that if you try to go and delete your newly-created history sub-folder, it will apparently go away, but then when you click on the green Done Editing button, and then return to the page, the history folder is still there. Ugh.

So, currently (as of Feb 4 thru 16 2019) there is no way I can see to delete files cleanly and completely, and to not have them interfere with new derivations -- using either the web interface nor the ia tool.

I can think of two options / operations to possibly add to the ia tool's delete sub-command:

a) add the ability to recursively delete a directory (-r / --recurse option), e.g.,

$ ia delete MYITEM history -r
or
$ ia delete MYITEM history/files -r

b) when deleting with ia, tell archive.org to please not even create a history folder in the first place ( -n / --nohistory option), e.g.,

$ ia delete MYITEM --all --nohistory

Wish I knew python better, I would try to help with this.

**Comment by vxbinaca** [comment]
I'm not at my computer right now so I can't access my VPS,  but the FAQ for the binary says how to avoid the history part.

**Comment by androclus** [comment]
are you referring to https://archive.org/services/docs/api/internetarchive/cli.html?highlight=history#delete where it says, "You can also prevent the backup from happening on deletes by adding -H x-archive-keep-old-version:0 to your command." ?

**Comment by vxbinaca** [comment]
Yes

**Comment by androclus** [comment]
Yeah, thanks. And good to know.  But a bit obscurely buried in the docs (what do S3 headers have to do with not doing a backup?, one would think.) When I did a "ia help delete", this would have been helpful here.

Besides, the real issues are that..

1) once the damage is done, the directories cannot be recursively deleted.

2) archive.org keeps picking up the files in the history (and thumbs/history) dirs and re-deriving them, (and making even further sub-dirs.. history/files/history/files..) and then serving the derived files (from the backed up ones) out to the public even (!) when they should have been ignored.

**Comment by vxbinaca** [comment]
Jakes working on v2 of internetarchive. It's coming. He's adding and fixing a lot of things.

**Comment by androclus** [comment]
#Yay  :-)

**Comment by saper** [comment]
I wonder if there is anything a client can do? This issue looks like a number of server-side issues. (I'd love to see a public bug tracker for these)

**Comment by vxbinaca** [comment]
Note: This issue was made for local directories not remote ones.

**Comment by samwilson** [comment]
There is now a `--no-backup` option:

```
    --no-backup                  Turn off archive.org backups. Clobbered files
                                 will not be saved to history/files/$key.~N~
                                 [default: True].
```
